geomesa-memory/src/main/scala/org/locationtech/geomesa/memory/cqengine/GeoCQEngine.scala

```diff
@@ -0,0 +1,75 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.memory.cqengine
+
+import java.util
+
+import com.googlecode.cqengine.query.option.DeduplicationStrategy
+import com.googlecode.cqengine.query.{QueryFactory, Query}
+import com.googlecode.cqengine.query.simple.All
+import org.locationtech.geomesa.memory.cqengine.utils.{CQEngineQueryVisitor, CQIndexingOptions}
+import org.locationtech.geomesa.utils.geotools._
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+import org.opengis.filter._
+
+class GeoCQEngine(sft: SimpleFeatureType) {
+  val cqcache = CQIndexingOptions.buildIndexedCollection(sft)
+
+  def remove(sf: SimpleFeature) {
```

Comment:
[minor] i think this syntax for unit methods is discouraged now - prefer `:Unit = `